.circleci/config.yml

```diff
@@ -32,20 +32,20 @@ jobs:
           command: "greenkeeper-lockfile-update"
 
       # Download and cache dependencies
-      - restore_cache:
-          keys:
-            - dependency-cache-{{ checksum "package.json" }}
-            # fallback to using the latest cache if no exact match is found
-            - dependency-cache-
+      #- restore_cache:
```

Comment:
Hmm, are you sure we want to disable the cache?

Reply:
Oh, just saw this comment:

> Disabling CI `node_modules` caching to solve failure after `truffle-hdwallet-provider` upgrade. (Refs: https://circleci.com/gh/tasitlabs/TasitSDK/593)

Reply:
Should we consider downgrading that package instead? Will we be able to re-enable caching at any point?

Reply:
We can downgrade `truffle-hdwallet-provider` package or try to re-enable in the future.

Reply:
What in the error message indicated to you that it was realted to `truffle-hdwallet-provider`?

```
> npx lerna bootstrap --hoist

lerna notice cli v3.13.4
lerna info ci enabled
lerna info Bootstrapping 6 packages
lerna info Installing external dependencies
lerna info hoist Installing hoisted dependencies into root
lerna info hoist Pruning hoisted dependencies
lerna info hoist Finished pruning hoisted dependencies
lerna ERR! npm install --no-save exited 1 in 'root'
lerna ERR! npm install --no-save stderr:
npm ERR! path /home/circleci/tasit-sdk/node_modules/web3-providers-ws/node_modules/websocket
npm ERR! code EISGIT
npm ERR! git /home/circleci/tasit-sdk/node_modules/web3-providers-ws/node_modules/websocket: Appears to be a git repo or submodule.
npm ERR! git     /home/circleci/tasit-sdk/node_modules/web3-providers-ws/node_modules/websocket
npm ERR! git Refusing to remove it. Update manually,
npm ERR! git or move it out of the way first.
```

Reply:
Actually, I wonder how much was actually being cached anyway. The way it was written now, it seems like it was only caching hoisted deps but probably not the node_modules libs in subdirs

Reply:
This is what we get from the latest truffle versions:

https://github.com/trufflesuite/truffle/releases/tag/v5.0.14
https://github.com/trufflesuite/truffle/releases/tag/v5.0.13
https://github.com/trufflesuite/truffle/releases/tag/v5.0.12
https://github.com/trufflesuite/truffle/releases/tag/v5.0.11
https://github.com/trufflesuite/truffle/releases/tag/v5.0.10
https://github.com/trufflesuite/truffle/releases/tag/v5.0.9

Reply:
Also, it looks like Truffle recommends upgrading by uninstalling truffle and then reinstalling it. Is that the way you updated it?

Reply:
> What in the error message indicated to you that it was realted to truffle-hdwallet-provider?

Almost sure that is the package because it's using that dependency ([refs](    "web3-provider-engine": "git+https://github.com/cgewecke/provider-engine.git#web3-one",))

> Actually, I wonder how much was actually being cached anyway. The way it was written now, it seems like it was only caching hoisted deps but probably not the node_modules libs in subdirs

1. With the `--hoist` option from `lerna` all dependencies are stored on the root `node_modules` dir. 
2. Maybe we need to use packages `package.json` files on the checksum command to only restore cache when none of these files was changed.
3. I've not noted any changes on the CI build run time.

Reply:
I think all of these subthreads are worth discussing, but I'm okay with disabling caching for now and creating an issue to reenable it.

Reply:
> Almost sure that is the package because it's using that dependency ([refs]( "web3-provider-engine": "git+https://github.com/cgewecke/provider-engine.git#web3-one",))

Makes sense.

> 1. With the --hoist option from lerna all dependencies are stored on the root node_modules dir.

I disagree. I think we've had this debate in the past? It only hoists ones that are used in at least 2 spots.

> 2. Maybe we need to use packages package.json files on the checksum command to only restore cache when none of these files was changed.

Yep, that's how it's supposed to be used.

> 3. I've not noted any changes on the CI build run time.

Oh okay, good to know. That's helpful, additional info in support of disabling caching being fine for now.

Reply:
>> Maybe we need to use packages package.json files on the checksum command to only restore cache when none of these files was changed.

> Yep, that's how it's supposed to be used.

Oh, or did you mean the child packages? In that case, yep, that's not just the default CircleCI recommendation config for non-lerna repos, and yes, I agree.

Reply:
> Oh okay, good to know. That's helpful, additional info in support of disabling caching being fine for now.

Ok, I'll create an issue.

> I disagree. I think we've had this debate in the past? It only hoists ones that are used in at least 2 spots.

Yes, I think we have talked about the `--hoist` before, but looking for my filesystem and reading the [docs](https://github.com/lerna/lerna/blob/master/doc/hoist.md) again, seems that only if a dependency with different versions will be installed on the local child package `node_modules`.

```sh
TasitSDK$ ls packages/*/node_modules
packages/tasit-account/node_modules:

packages/tasit-action/node_modules:
tasit-account  tasit-contracts

packages/tasit-contract-based-account/node_modules:
tasit-action  tasit-contracts

packages/tasit-contracts/node_modules:

packages/tasit-link-wallet/node_modules:
tasit-action

packages/tasit-sdk/node_modules:
tasit-account  tasit-action  tasit-contract-based-account  tasit-contracts

TasitSDK$ ls node_modules/ | wc -l
787
```

Reply:
> packages/tasit-action/node_modules:
> tasit-account  tasit-contracts

This one is surprising then, right?

Reply:
The code you shared is pretty convincing - I think lerna's docs are wrong, then. Because it says that it hoists "shared" dependencies.

Reply:
I made this issue on the lerna repo https://github.com/lerna/lerna/issues/2062

Reply:
> > packages/tasit-action/node_modules:
> > tasit-account  tasit-contracts
> 
> This one is surprising then, right?

I didn't get it. `tasit-action` is using `tasit-contracts` to resolve contracts ABI/address and `tasit-account` as devDependency to access `createFromPrivateKey` helper.

Reply:
The issue to enable cache in future: https://github.com/tasitlabs/TasitSDK/issues/359

Reply:
> > > packages/tasit-action/node_modules:
> > > tasit-account  tasit-contracts
> > 
> > 
> > This one is surprising then, right?
> 
> I didn't get it. `tasit-action` is using `tasit-contracts` to resolve contracts ABI/address and `tasit-account` as devDependency to access `createFromPrivateKey` helper.

Yep. So why aren't they hoisted like all of our other deps?